src/libstd/net/tcp.rs

```diff
@@ -498,18 +498,45 @@ impl TcpStream {
 
     /// Moves this TCP stream into or out of nonblocking mode.
     ///
-    /// On Unix this corresponds to calling fcntl, and on Windows this
-    /// corresponds to calling ioctlsocket.
+    /// This will result in `read`, `write`, `recv` and `send` operations
+    /// becoming nonblocking, i.e. immediately returning from their calls.
+    /// If the IO operation is successful, `Ok` is returned and no further
+    /// action is required. If the IO operation is pending, an error with kind
```

Comment:
Saying that the operation is pending might be a bit misleading - `WouldBlock` means that the request couldn't be satisfied at that moment and you need to retry later. It's not the case that the operation's been queued up and will eventually finish.

Reply:
thanks for mentioning this, was a little confused how it worked behind the scenes. i addressed it in https://github.com/rust-lang/rust/pull/45227/commits/6c20fe50fd4180ce89f15b2fac93cc580747d1d8 but let me know if you know a better way of wording it